src/test/ui/consts/control-flow/interior-mutability.rs

```diff
@@ -0,0 +1,18 @@
+#![feature(const_if_match)]
+
+use std::cell::Cell;
+
+// Ensure that *any* assignment to the return place of a value with interior mutability
+// disqualifies it from promotion.
+const X: Option<Cell<i32>> = {
+    let mut x = None;
+    if false {
+        x = Some(Cell::new(4));
+    }
+    x
+};
```

Comment:
Would be good to repeat this for `if let` and `match` (again to not presume the MIR lowering).

Reply:
Did you miss this one?

Reply:
If I marked it resolved, it was by accident. I don't think it adds anything to repeat each qualifs test three times, however.

Reply:
I agree. If the lowerings start differing in ways that are problematic here we're gonna see other problems, too.